### PR TITLE
PRO-1144 archive choices

### PR DESCRIPTION
**Pull request by stuartromanek**
https://linear.app/apostrophecms/issue/PRO-1144/implement-the-archive-confirmation-modal

designs: https://www.sketch.com/s/1663b37a-402f-4441-9c48-2e907707bf32/p/pages-organize

### CODE REVIEW

File: modules/@apostrophecms/page-type/index.js [hunk 1]
```diff
@@ -392,9 +392,6 @@ module.exports = {
   },
   extendMethods(self) {
     return {
-      enableAction() {
```
Comment (boutell) [inline modules/@apostrophecms/page-type/index.js]: Why the change back?

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/page-type/index.js (hunk 1 above)
Comment (boutell) [reply modules/@apostrophecms/page-type/index.js]: (Changing this back is not a blocker for me.)

Comment (stuartromanek) [reply modules/@apostrophecms/page-type/index.js]: I wasn't sure we were settled on this approach. I [suggested coming at this from the other direction here](https://github.com/apostrophecms/apostrophe/pull/2928#issuecomment-823262175), but if we like inheriting `page`'s action I'll push ahead in the other PR with the suggested cleanup

File: modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js [hunk 2]
```diff
@@ -8,32 +8,97 @@ export default {
...
-            const archiveId = pageTree._children.filter(p => p.type === '@apostrophecms/archive-page')[0]._id;
-            body._targetId = archiveId;
+            body._targetId = '_archive';
             body._position = 'lastChild';
+
+            if (confirm.data && confirm.data.choice === 'this') {
+              // Editor wants to archive one page but not it's children
+              // Before archiving the page in question, move the children up a level,
+              // preserving their current order
+              doc._children.forEach(async child => {
```
[the remark this answers is not in the document]
Comment (stuartromanek) [reply modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js:75]: I knew this was a thing i never did before ..

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue [hunk 3]
```diff
@@ -568,7 +568,31 @@ export default {
...
-        this.modal.showModal = false;
+        if (!this.restoreOnly) {
+          this.modal.showModal = false;
+        }
+        if (this.restoreOnly) {
+          apos.bus.$emit('busy', {
+            name: 'busy',
+            active: true
+          });
+          setTimeout(async () => {
```
[the remark this answers is not in the document]
Comment (stuartromanek) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:579]: purely a UX chain of events timing delay but I might approach this differently now that i'm thinking about it again

File: modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js [hunk 4]
```diff
@@ -8,32 +8,97 @@ export default {
     // appropriately, not returned or thrown to the caller.
     //
     // Returns `true` if the document was ultimately archived.
-    async archive(action, _id, isPublished, isPage) {
+
+    async archive(doc, isPage) {
```
[the remark this answers is not in the document]
Comment (stuartromanek) [reply modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js:12]: I'm open to tidying .. what would you suggest?

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (boutell) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:579]: It's OK with me as long as it's not relying on it for a functional reason.

File: modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js:12]: `doc.slug.startsWith('/')` is a reliable test of page-iness.

File: modules/@apostrophecms/page-type/index.js (hunk 1 above)
Comment (boutell) [reply modules/@apostrophecms/page-type/index.js]: Sure, a separate PR would be fine.

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (stuartromanek) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:592]: i did this a lil differently but the timeout is still there, the modals just dont open and close during the busy

File: modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (stuartromanek) [reply modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js]: resolved

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue [hunk 5]
```diff
@@ -580,7 +590,8 @@ export default {
             active: true
           });
           setTimeout(async () => {
-            this.modal.showModal = false;
+            this.filters.archived = false;
+            this.loadDoc();
```
Comment (boutell) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:594]: Hang on... so you mark it busy, then you wait a full second before doing anything, then you load the document but you don't await it, and then you immediately mark it not busy. Does that make sense?

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue (hunk 5 above)
Comment (stuartromanek) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue]: No, some of that is a mistake, but also it's complicating the UX. Got rid of the timeout and just re populate the doc with the freshly published one.

**Review by boutell — APPROVED** [review]
